marketplace-client/src/marketplace/src/app/account/_components/password.form.component.ts

```diff
@@ -0,0 +1,77 @@
+import {Component, Input, Output, EventEmitter} from '@angular/core';
+import {AbstractControl, FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {AccountService} from '../../_services/account.service';
+import {first} from 'rxjs/operators';
+import {ActivatedRoute, Router} from '@angular/router';
+import {validateConfirmPassword, validatePassword} from '../../_helpers/validators.service';
+
+@Component({
+  selector: 'app-password-form',
+  templateUrl: './password.form.component.html',
+})
+
+export class PasswordFormComponent {
+
+  @Output() onClick  = new EventEmitter();
+
+  form: FormGroup;
+
+  // icons
+  loading = false;
+  submitted = false;
+  showPassword = false;
+  showConfirmPassword = false;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private accountService: AccountService,
+    private route: ActivatedRoute,
+  ) {
+    this.form = this.formBuilder.group({
+      password: ['', [Validators.required, Validators.minLength(6)]],
+      confirmPassword: ['', Validators.required],
+    }, {
+      validator: [validateConfirmPassword, validatePassword]
+    });
+  }
+
+  get getForm(): { [p: string]: AbstractControl } { return this.form.controls; }
+
+  onSubmit(): void {
+    this.submitted = true;
+    if (this.form.invalid) {
+      return;
+    }
+    this.loading = true;
+    this.form.disable();
+    this.route.queryParamMap.subscribe(params => {
+      const id = params.get('link');
+      console.log(id);
+      if (id) {
+        this.accountService.setNewPassword(id, this.form.get('password')?.value)
+          .pipe(first())
+          .subscribe({
```

Comment:
`subscribe` inside of the `subscribe` (moreover, without unsubscriptions) is a source of memory leaks and unexpected async errors. Leverage the RxJS flattening operators:

https://ramya-bala221190.medium.com/simple-example-for-rxjs-flattening-operators-2456628e4c07

http://alanpryorjr.com/2019-05-15-rxjs-flattening-operators/#:~:text=In%20reality%2C%20the%20flattening%20operators,observable%20that%20is%20emitting%20values.&text=Contrast%20this%20with%20map%20%2C%20which,value%20(not%20another%20observable).